lib/pbench/test/unit/server/test_generate_key.py

```diff
@@ -0,0 +1,38 @@
+from http import HTTPStatus
+
+import pytest
+import requests
+
+
+class TestDatasetsAccess:
+    @pytest.fixture()
+    def query_get_as(self, client, server_config):
+        """
+        Helper fixture to perform the API query and validate an expected
+        return status.
+
+        Args:
+            client: Flask test API client fixture
+            server_config: Pbench config fixture
+        """
+
+        def query_api(user_token, expected_status: HTTPStatus) -> requests.Response:
+            headers = {"authorization": f"bearer {user_token}"}
+            response = client.post(
+                f"{server_config.rest_uri}/generate_key",
+                headers=headers,
+            )
+            assert response.status_code == expected_status
+            return response
+
+        return query_api
+
+    def test_unauthorized_access(self, query_get_as, pbench_drb_token_invalid):
+        response = query_get_as(pbench_drb_token_invalid, HTTPStatus.UNAUTHORIZED)
+        assert response.json == {
+            "message": "User provided access_token is invalid or expired token"
+        }
```

Comment:
What are we actually testing here?  I think we're supposed to be testing `/generate_key`, but the message is referring to an _`access_token`_ not an API key.

Is `pbench_drb_token_invalid` the correct fixture?  (Shouldn't we be using `pbench_drb_api_key_invalid`?)

A similar comment applies to `test_successful_api_key_generation()`.

Reply:
> What are we actually testing here? I think we're supposed to be testing `/generate_key`, but the message is referring to an _`access_token`_, not an API key.

Here we are testing whether the user is an authorized user or not (i.e) If a `user` hits this API with invalid `access_token` means, then the user won't be able to create an `api_key`

>Is pbench_drb_token_invalid the correct fixture? (Shouldn't we be using pbench_drb_api_key_invalid?)

Yes because `pbench_drb_api_key_invalid` will return a created `api_key` which is `expired` .(But I agree that ,we can also use `pbench_drb_api_key_invalid` fixture as well for authenticating it)

Reply:
OK, I understand now.  (I made the above comment at the end of a long review and I was basically "out of gas".)

So, yes, this module tests the `/generate_key` API using _access tokens_ which are either valid or invalid...that makes good sense.  (There should be no need to test using the `pbench_drb_api_key_*` fixtures, because the application or effect of those aren't specific to the `/generate_key` API.)